src/gui/newwizard/states/accountconfiguredsetupwizardstate.cpp

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) Fabian Müller <fmueller@owncloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#include <QDir>
+
+#include "accountconfiguredsetupwizardstate.h"
+#include "gui/folderman.h"
+#include "pages/accountconfiguredwizardpage.h"
+
+namespace OCC::Wizard {
+
+AccountConfiguredSetupWizardState::AccountConfiguredSetupWizardState(SetupWizardContext *context)
+    : AbstractSetupWizardState(context)
+{
+    // being pessimistic by default
+    bool vfsIsAvailable = false;
+    bool enableVfsByDefault = false;
+    bool vfsModeIsExperimental = false;
+
+    switch (bestAvailableVfsMode()) {
+    case Vfs::WindowsCfApi:
+        vfsIsAvailable = true;
+        enableVfsByDefault = true;
+        vfsModeIsExperimental = false;
+        break;
+    case Vfs::WithSuffix:
+        vfsIsAvailable = true;
+        enableVfsByDefault = false;
+        vfsModeIsExperimental = true;
+        break;
+    default:
+        break;
+    }
+
+    _page = new AccountConfiguredWizardPage(FolderMan::suggestSyncFolder(_context->accountBuilder().serverUrl(), _context->accountBuilder().displayName()), vfsIsAvailable, enableVfsByDefault, vfsModeIsExperimental);
+}
+
+SetupWizardState AccountConfiguredSetupWizardState::state() const
+{
+    return SetupWizardState::AccountConfiguredState;
+}
+
+void AccountConfiguredSetupWizardState::evaluatePage()
+{
+    auto accountConfiguredSetupWizardPage = qobject_cast<AccountConfiguredWizardPage *>(_page);
+    OC_ASSERT(accountConfiguredSetupWizardPage != nullptr);
```

Comment:
```suggestion
    Q_ASSERT(accountConfiguredSetupWizardPage != nullptr);
```

Reply:
There is no way how to continue here. The client will crash in the next line. This situation should *never* ever occur.

Reply:
then OC_ENFORCE